internal/provider/data_source_gitlab_project_milestones_test.go

```diff
@@ -0,0 +1,40 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataGitlabProjectMilestones_basic(t *testing.T) {
+	testAccCheck(t)
+
+	testProject := testAccCreateProject(t)
+	testMilestones := testAccAddProjectMilestones(t, testProject, 2)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataGitlabProjectMilestonesConfig(testProject.ID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.gitlab_project_milestones.this", "milestones.#", fmt.Sprintf("%d", len(testMilestones))),
+					resource.TestCheckResourceAttr("data.gitlab_project_milestones.this", "milestones.0.title", testMilestones[1].Title),
+					resource.TestCheckResourceAttr("data.gitlab_project_milestones.this", "milestones.0.description", testMilestones[1].Description),
+					resource.TestCheckResourceAttr("data.gitlab_project_milestones.this", "milestones.1.title", testMilestones[0].Title),
+					resource.TestCheckResourceAttr("data.gitlab_project_milestones.this", "milestones.1.description", testMilestones[0].Description),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataGitlabProjectMilestonesConfig(projectID int) string {
```

Comment:
Move this inline to adhere to the upcoming testing guardrails: https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008